src/main/java/projectbookstore/service/impl/BookServiceImpl.java

```diff
@@ -41,10 +41,9 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto update(Long id, CreateBookRequestDto requestDto) {
-        bookRepository.findById(id).orElseThrow(
-                () -> new EntityNotFoundException(
-                String.format("Can`t find a book with id: %d", id)
-        ));
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can`t find a book with id: " + id);
+        }
```

Comment:
this could also be a private method used in update & delete to avoid code duplication